src/main/java/com/cloudbees/jenkins/support/filter/ContentMappings.java

```diff
@@ -121,14 +121,13 @@ private ContentMappings(@Nonnull XmlProxy proxy) {
      * Looks up or creates a new ContentMapping for the given original string and a ContentMapping generator.
      */
     public @Nonnull ContentMapping getMappingOrCreate(@Nonnull String original, @Nonnull Function<String, ContentMapping> generator) {
-        return mappings.computeIfAbsent(original, generator.andThen(mapping -> {
-            try {
-                save();
-            } catch (IOException e) {
-                LOGGER.log(Level.WARNING, "Could not save mappings file", e);
-            }
-            return mapping;
-        }));
+        ContentMapping mapping = mappings.computeIfAbsent(original, generator);
+        try {
+            save();
```

Comment:
Previously, this was called before the new mapping was actually in the map. I think it doesn't actually cause an issue in standard usage, but before wrapping everything in `BulkChange` this would keep the last mapping added from being saved. I only noticed it because I was adding a serialization round trip test, but I think it is good to fix it anyway in case the method ends up being used outside of `BulkChange` in the future.